mpv.py

```diff
@@ -512,6 +518,29 @@ def playlist_next(self, mode='weak'):
     def playlist_prev(self, mode='weak'):
         self.command('playlist_prev', mode)
 
+    @property
+    def playlist_position(self):
+        val = self._get_property('playlist-pos')
+        return int(val) if val else None
+
+    @property
+    def playlist_length(self):
+        return self._get_property('playlist-count', proptype=int)
+
+    @playlist_position.setter
+    def playlist_position(self, new):
+        if new >= self.playlist_length or new < 0:
+            raise ValueError("Trying to access out of bounds index %d in mpv playlist" % new)
+        else:
+            self._set_property("playlist-pos", new, proptype=int)
+
+    @property
+    def filenames():
+        res = []
+        for i in range(self.playlist_length):
+            self._get_property("playlist/{}/filename".format(i), proptype=str)
+        return res
```

Comment:
How about
```python
return [ e['filename'] for e in self.playlist ]
```
?

The ```proptype=str``` is a big no-no here as paths are not guaranteed to be in *any* sane encoding - thus this would throw an encoding error or return garbage if the playlist contains a file with a weird name, e.g. a file from a windows file system accessed from a linux machine.

In case the API user wants to handle paths as strings they can do that decoding and error handling explicitly.